python-api/tests/integrationtests/test_controller.py

```diff
@@ -249,6 +251,82 @@ def test_get_preview_ports(self):
                     print log.read()
 
 
+class TestSignals(object):
+
+    """Test on_*() methods"""
+
+    def run_mainloop(self):
+        self.mainloop = GLib.MainLoop()
+        self.mainloop.run()
+
+
+    def quit_mainloop(self, *args):
+        self.mainloop.quit()
+
+    def test_on_new_mode_online(self):
+        """Create a Controller object, call on_new_mode_online method and check that the callback fires"""
+        serv = Server(path=PATH)
+        try:
+            serv.run()
+
+            controller = Controller()
+            controller.establish_connection()
+
+            test_cb = Mock(side_effect=self.quit_mainloop)
+            controller.on_new_mode_online(test_cb)
+            controller.set_composite_mode(0)
+
+            GLib.timeout_add_seconds(5, self.quit_mainloop)
```

Comment:
`test_cb = Mock(side_effect=self.quit_mainloop)` quits the Mainloop as soon as the callback arrives.

If no callback arrives, the timeout kills it after max. 5 seconds.